ssl/statem/extensions_cust.c

```diff
@@ -225,6 +225,9 @@ int custom_ext_add(SSL_CONNECTION *s, int context, WPACKET *pkt, X509 *x,
                 || !WPACKET_start_sub_packet_u16(pkt)
                 || (outlen > 0 && !WPACKET_memcpy(pkt, out, outlen))
                 || !WPACKET_close(pkt)) {
+            if (meth->free_cb != NULL)
+                meth->free_cb(SSL_CONNECTION_GET_SSL(s), meth->ext_type, context,
+                              out, meth->add_arg);
```

Comment:
Shouldn't we do the same thing after the `ossl_assert` below?

Reply:
it is an assert so something that should not be possible?

Reply:
Right, but in a non-debug build ossl_assert still gets compiled and the failure path will be taken if ever the assertion turns out not to be true. We should still cleanup from such failures even after ossl_assert.